### PR TITLE
Remove ONNX from requirements.txt

### PR DESCRIPTION
**Pull request by prabhat00155**
**Description**: Remove ONNX package from requirements.txt, making it optional when using onnxruntime.

**Motivation and Context**
- Why is this change required? What problem does it solve? You can use onnxruntime to do inferencing without ONNX package installed. Having ONNX as a dependency causes issue on aarch devices as ONNX whl is not there for aarch64 platform and has to be built from source, which has a bunch of dependencies and is very slow on these devices with limited compute.
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 numpy >= 1.16.6
-onnx >= 1.2.3
```
Comment (jywu-msft) [inline requirements.txt:2]: @pranavsharma , @snnn , @faxu and I had a discussion and we decided we want to remove the onnx requirement entirely (not just for aarch64)
the onnxruntime python api has no dependency on onnx python package.
it's only needed for onnx backend api, which we will support 'unofficially'

Comment (prabhat00155) [reply requirements.txt:2]: Makes sense.

Comment (pranavsharma) [reply requirements.txt:2]: Can you also please add a note in backend.py that this is not the official Python API? We'd also discussed removing the mention of the backend API here https://microsoft.github.io/onnxruntime/python/api_summary.html#backend. Can be in a different PR.

Comment (prabhat00155) [reply requirements.txt:2]: Added the note.

**Comment by jywu-msft** [comment]
nGraph and Android CI failures are existing known failures.

**Pull request merged**